### ISSUE TITLE
Command line parameters

### ISSUE THREAD

**Issue by andrearastelli** [raytracingDevTeam/raytracing#33]
To simplify the execution of the application, and to avoid merge issues, we should provide of a way to set the input parameters through command line arguments.

This has to be done before releasing V1.0.0

**Comment by andrearastelli** [comment]
@Ale32 The command line arguments are now defined.

The functionality is not fully tested, so if something weird happen, has to be fixed.

The base parameter accepted for now are:
- output: The output path of the image. The path must exists, or the image will not be created.
- width: The width of the resulting image, default 200px.
- height: The height, default 100px.
- samples: The number of samples to use, default 8.

**Comment by Ale32** [comment]
I thought there was a standard way to do this, so it's better for us to use and mantain.

Do you think that this [Boost.Program_options](https://www.boost.org/doc/libs/1_67_0/doc/html/program_options.html) is too much for what we want to achieve?

**Comment by andrearastelli** [comment]
Is not too much, but require us to use boost.
Do we want to use an external library?

I don't feel that we should focus on such features, right now.

We can always improve the way the command line arguments are parsed, so we can open a new issue referring that, but for the time being I think we should focus on raytracing features.

What do you think?

By the way, Boost will help with other stuff as well, but will add lots of dependencies that I'm not sure we want.
For example, in PBRT, the solution to command line parameters was to write their own parser, so the raytracer has no external dependencies (except the very basic ones, like OpenEXR).

**Comment by Ale32** [comment]
That's right, the less are the dependencies the better is.

**Comment by andrearastelli** [comment]
Will merge #34 tonight then
